packages/flutter/lib/src/cupertino/dialog.dart

```diff
@@ -766,33 +1338,59 @@ class _RenderCupertinoDialog extends RenderBox {
   }
 
   @override
-  bool hitTestChildren(BoxHitTestResult result, { required Offset position }) {
+  bool hitTestChildren(BoxHitTestResult result, {required Offset position}) {
+    if (isActionSheet) {
+      final MultiChildLayoutParentData contentSectionParentData = contentSection!.parentData! as MultiChildLayoutParentData;
+      final MultiChildLayoutParentData actionsSectionParentData = actionsSection!.parentData! as MultiChildLayoutParentData;
+      return result.addWithPaintOffset(
+            offset: contentSectionParentData.offset,
+            position: position,
+            hitTest: (BoxHitTestResult result, Offset transformed) {
+              assert(transformed == position - contentSectionParentData.offset);
+              return contentSection!.hitTest(result, position: transformed);
+            },
+          ) ||
+          result.addWithPaintOffset(
+            offset: actionsSectionParentData.offset,
+            position: position,
+            hitTest: (BoxHitTestResult result, Offset transformed) {
+              assert(transformed == position - actionsSectionParentData.offset);
+              return actionsSection!.hitTest(result, position: transformed);
+            },
+          );
+    }
+
     final BoxParentData contentSectionParentData = contentSection!.parentData! as BoxParentData;
     final BoxParentData actionsSectionParentData = actionsSection!.parentData! as BoxParentData;
     return result.addWithPaintOffset(
-             offset: contentSectionParentData.offset,
-             position: position,
-             hitTest: (BoxHitTestResult result, Offset transformed) {
-               assert(transformed == position - contentSectionParentData.offset);
-               return contentSection!.hitTest(result, position: transformed);
-             },
-           )
+            offset: contentSectionParentData.offset,
+            position: position,
+            hitTest: (BoxHitTestResult result, Offset transformed) {
+              assert(transformed == position - contentSectionParentData.offset);
+              return contentSection!.hitTest(result, position: transformed);
+            },
+          )
         || result.addWithPaintOffset(
-             offset: actionsSectionParentData.offset,
-             position: position,
-             hitTest: (BoxHitTestResult result, Offset transformed) {
-               assert(transformed == position - actionsSectionParentData.offset);
-               return actionsSection!.hitTest(result, position: transformed);
-             },
-           );
+            offset: actionsSectionParentData.offset,
+            position: position,
+            hitTest: (BoxHitTestResult result, Offset transformed) {
+              assert(transformed == position - actionsSectionParentData.offset);
+              return actionsSection!.hitTest(result, position: transformed);
+            },
+          );
```

Comment:
I see there was an indentation change here, too.  I can't tell from this diff if the closing parenthesis here is aligned with `result.addWithPaintOffset`, but that's how it should be.  If it is, then this change is fine.